dex/elrond_dex_farm/src/lib.rs

```diff
@@ -298,23 +300,43 @@ pub trait Farm:
             self.decrease_reward_reserve(&reward)?;
         }
 
-        let new_attributes = FarmTokenAttributes {
+        let new_initial_farming_amount = self.rule_of_three_non_zero_result(
+            &amount,
+            &farm_attributes.current_farm_amount,
+            &farm_attributes.initial_farming_amount,
+        )?;
+        let new_compound_reward_amount = self.rule_of_three(
+            &amount,
+            &farm_attributes.current_farm_amount,
+            &farm_attributes.compounded_reward,
+        );
+
+        let mut new_attributes = FarmTokenAttributes {
             reward_per_share: self.reward_per_share().get(),
             entering_epoch: farm_attributes.entering_epoch,
             apr_multiplier: farm_attributes.apr_multiplier,
             with_locked_rewards: farm_attributes.with_locked_rewards,
+            initial_farming_amount: new_initial_farming_amount,
+            compounded_reward: new_compound_reward_amount,
+            current_farm_amount: amount.clone(),
         };
 
         let caller = self.blockchain().get_caller();
         self.burn_farm_tokens(&payment_token_id, token_nonce, &amount)?;
-        let new_nonce = self.create_farm_tokens(&amount, &farm_token_id, &new_attributes);
+        let mut farm_amount = amount.clone();
+        let new_nonce = self.create_farm_tokens_by_merging(
+            &mut farm_amount,
+            &farm_token_id,
+            &mut new_attributes,
+            &caller,
+        )?;
         self.send_nft_tokens(
             &farm_token_id,
             new_nonce,
             &amount,
```

Comment:
Shouldn't this be sending `farm_amount` (i.e. the merged amount) instead of `amount`?

Reply:
True, but since it's already fixed in this branch: https://github.com/ElrondNetwork/sc-dex-rs/pull/106 we'll fix it when we merge this one too